src/main/java/org/cyclonedx/generators/xml/AbstractBomXmlGenerator.java

```diff
@@ -79,413 +89,36 @@ public DocumentBuilder buildSecureDocumentBuilder() throws ParserConfigurationEx
-        if (contacts != null && !contacts.isEmpty()) {
-            final Element contactsNode = createElement(parent, parentElementName);
-            for (OrganizationalContact contact : contacts) {
-                createOrganizationalContactNode(contactsNode, contact, elementName);
-            }
-        }
-    }
-
-    private void createOrganizationalContactNode(Node parent, OrganizationalContact contact, String elementName) {
-        final Element contactNode = createElement(parent, elementName);
-        createElement(contactNode, "name", stripBreaks(contact.getName()));
-        createElement(contactNode, "email", stripBreaks(contact.getEmail()));
-        createElement(contactNode, "phone", stripBreaks(contact.getPhone()));
-        processExtensions(contactNode, contact);
-    }
-
-    private void createOrganizationalEntityNode(Node parent, OrganizationalEntity entity, String elementName) {
-        if (entity != null) {
-            final Element entityNode = createElement(parent, elementName);
-            createElement(entityNode, "name", stripBreaks(entity.getName()));
-            createElement(entityNode, "url", stripBreaks(entity.getUrl()));
-            if (entity.getContacts() != null && entity.getContacts().size() > 0) {
-                createOrganizationalContactNode(entityNode, entity.getContacts().get(0), "contact");
-            }
-            processExtensions(entityNode, entity);
-        }
-    }
+    protected Document generateDocument(final Bom bom)
+        throws ParserConfigurationException
+    {
+        try {
+            final DocumentBuilder docBuilder = buildSecureDocumentBuilder();
 
-    protected void processExtensions(final Node node, final ExtensibleElement extensibleElement) {
-        if (extensibleElement != null && extensibleElement.getExtensibleTypes() != null) {
-            for (final ExtensibleType t: extensibleElement.getExtensibleTypes()) {
-                final Element e = createElementNS(node, t.getNamespace(), t.getName());
-                if (t.getAttributes() != null) {
-                    for (Attribute attr: t.getAttributes()) {
-                        e.setAttribute(attr.getKey(), attr.getValue());
-                    }
-                }
-                if (t.getValue() != null) {
-                    e.appendChild(doc.createTextNode(t.getValue()));
-                } else if (t.getExtensibleTypes() != null) {
-                    for (final ExtensibleType childType: t.getExtensibleTypes()) {
-                        final Element childNode = createElementNS(e, childType.getNamespace(), childType.getName());
-                        processExtensions(childNode, childType);
-                    }
-                }
-            }
-        }
-    }
+            this.doc = docBuilder.parse(
+                new InputSource(
+                    new StringReader(
+                        toXML(bom))));
```

Comment:
Personal preference but this look better in on line

Reply:
76c095e